### DISCUSSION TITLE
Hiding auto-inferred props in addon-controls

### DISCUSSION THREAD

**Discussion by thany** [storybookjs/storybook#17236] [Help]
I have a situation where I have a specific set of props defined in `args` & `argTypes` and I don't need to see any of the automatically inferred ones. But what actually happens, is that I get to see both.

How can I disable automatically inferred args for a single story? I prefer not to write it like this:

```js
Story.argTypes = {
  specialProp: {
    control: { type: 'range', min: 0, max: 10 },
  },
  prop1: { control: { type: 'disabed' } },
  prop2: { control: { type: 'disabed' } },
  prop3: { control: { type: 'disabed' } },
};
```

That's a bit too un-DRY to me, and it also "breaks" when adding more props, and forgetting to disable them too.

Rather I'm looking for something generic:

```js
Story.argTypes = {
  specialProp: {
    control: { type: 'range', min: 0, max: 10 },
  },
  '*': { control: { type: 'disabled' } },
};
```

*(this doesn't work; it just serves as a proposal)*

I couldn't find this either [here](https://storybook.js.org/docs/react/essentials/controls), nor [here](https://storybook.js.org/docs/react/writing-stories/args). (two different documentations for what seems to be the same feature, is that right?)
Neither of these appear to provide a full comprehensive reference to `args` or `argTypes`. Only the former one mentions auto inferrence very subtly, which makes it look to me like kind of a hidden feature. Very cool, but not for this one story I've got 🙂

Alternatively, it would be helpful if automatic inference wouldn't do anything as soon as custom `args` are used. Not sure if this is the best solution though, because I can see the advantage of having both in some cases.

**Comment by YassienW** [comment]
Here's what I've done:

```tsx
export const Main: Story = {
  args: {
    prop1: '#ffffff',
    prop2 200,
    prop3: 100,
  }
};
Main.parameters = {
  controls: {include: Object.keys(Main.args)}
}
```

This includes only the controls that have been defined in args, you could use argTypes if you wish
